lib_nbgl/src/nbgl_lns.c

```diff
@@ -0,0 +1,542 @@
+            if (screen_ctx.pagination_pages == 1) {
+                screen_ctx.line2 = screen_content.subtext;
+            }
+        }
+        else {
+            // Generate title with pagination
+            // Consider the resulting text fit in screen
+            char buffer[SCREEN_LINE_MAX_STRING_LEN];
+
+#if 1
+            // Spare flash by not using snprintf here
+            char page[4];
+
+            strlcpy(buffer, screen_content.text, sizeof(buffer));
+            strlcat(buffer, " (", sizeof(buffer));
+            u8toa(screen_ctx.pagination_current_page, page);
+            strlcat(buffer, page, sizeof(buffer));
+            strlcat(buffer, "/", sizeof(buffer));
+            u8toa(screen_ctx.pagination_pages, page);
+            strlcat(buffer, page, sizeof(buffer));
+            strlcat(buffer, ")", sizeof(buffer));
+#else
+            snprintf(buffer,
+                     sizeof(buffer),
+                     "%s (%d/%d)",
+                     screen_content.text,
+                     screen_ctx.pagination_current_page,
+                     screen_ctx.pagination_pages);
+#endif
+
+            line1 = buffer;
+
+            const char *start;
+            uint8_t     len;
+
+            nbgl_font_compute_paging(screen_content.subtext,
+                                     screen_ctx.pagination_current_page,
+                                     lines_width,
+                                     false,
+                                     &start,
+                                     &len);
+            screen_ctx.line2     = start;
+            screen_ctx.line2_len = len;
+        }
+    }
+
+    if (line1 != NULL) {
+        display_string_with_len(
+            line1, line1_len, screen_content.bold, screen_content.centered, line1_x, line1_y);
+    }
+    return true;
+}
+
+static bool draw_line_2(void)
+{
+    if (screen_ctx.line2 == NULL) {
+        return false;
+    }
+
+    uint8_t text_x = TEXT_X_OFFSET;
+    uint8_t text_y = TEXT_Y_OFFSET_SECOND_LINE;
+    if (screen_content.icon != NULL) {
+        // icon present, text is shifted to right to give space to the icon
+        text_x = TEXT_X_OFFSET_BESIDE_ICON;
+    }
+    display_string_with_len(screen_ctx.line2,
+                            screen_ctx.line2_len,
+                            screen_ctx.line2_bold,
+                            screen_content.centered,
+                            text_x,
+                            text_y);
+    return true;
+}
+
+static bool draw_left_nav(void)
+{
+    bool                left = false;
+    nbgl_stepPosition_t pos  = screen_content.pos & STEP_POSITION_MASK;
+
+    if ((screen_ctx.pagination_pages > 1) && (screen_ctx.pagination_current_page > 1)) {
+        left = true;
+    }
+
+    if ((pos == NEITHER_FIRST_NOR_LAST_STEP) || (pos == LAST_STEP)) {
+        left = true;
+    }
+
+    if (left) {
+        if (screen_content.vertical_nav) {
+            display_glyph(&C_icon_up, 2, 15, 7, 4);
```

Comment:
Would be helpful to have defines, in order to follow easier the coce, and adapt if needed later?